src/pages/notifications/index.vue

```diff
@@ -301,26 +305,28 @@ export default {
         // No-op
       }
     }
+    this.$api.$post(updateEventLastSeen());
+    this.lastEntryTs = Date.now();
     window.addEventListener('beforeunload', this.updateEventLastSeenTs);
   },
   // For SPA navigation
   beforeRouteLeave(to, from, next) {
-    this.updateEventLastSeenTs();
-    window.removeEventListener('beforeunload', this.updateEventLastSeenTs);
+    this.handleRouteLeave();
+    window.removeEventListener('beforeunload', this.handleRouteLeave);
```

Comment:
Since event list won't update unless you press refresh, `updateEventLastSeenTs` only needs to be called on mounted and on refresh? Otherwise on leave it might cause unseen even in between treated as seen